### ISSUE TITLE
HF parity gate compares golden logits against themselves (vacuous truth)

### ISSUE THREAD

**Issue by noahgift** [paiml/apr-model-qa-playbook#18]
## Description

`gates.rs:333` — the HF parity test compares golden logits with themselves:

```rust
// Run inference to get actual logits
// For now, we do a self-consistency check (golden vs golden)
let result = oracle.tensors_close(&golden.logits, &golden.logits);
```

This always passes because `x == x` is always true. The test provides zero falsification value.

## Impact

P1 — Any model passes HF parity regardless of actual output. The gate exists but provides no quality signal. Violates Popperian falsification — a test that cannot fail is not a test.

## Expected Behavior

Should compare model inference output against the golden reference logits.

## Relation

Related to #13 (HfParityOracle vacuous cache). Both issues stem from HF parity being a stub implementation.

## Found By

Popperian falsification code review (round 20).
